### PR TITLE
[Broker] Change create topic return error to Status.BAD_REQUEST

### PR DESCRIPTION
**Pull request by congbobo184**
### Motivation
When create transaction system topic, broker will return Status.CONFLICT, It does not conform to the wrong return semantics of the method. So change it to Status.BAD_REQUEST, when client receive Status.BAD_REQUEST, it will generic PulsarAdminException and error code is Status.BAD_REQUEST

### Modifications
change the error code to Status.BAD_REQUEST when create transaction system topic.

### Verifying this change
change test verify

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -250,7 +250,7 @@ protected void validateAdminAndClientPermission() {
     protected void validateCreateTopic(TopicName topicName) {
         if (isTransactionInternalName(topicName)) {
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:251]: The topic `TRANSACTION_COORDINATOR_ASSIGN` could be created by users, right?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -250,7 +250,7 @@ protected void validateAdminAndClientPermission() {
...
             log.warn("Try to create a topic in the system topic format! {}", topicName);
-            throw new RestException(Status.CONFLICT, "Cannot create topic in system topic format!");
+            throw new RestException(Status.BAD_REQUEST, "Cannot create topic in system topic format!");
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:253]: Could we list the system topics here? Users could know which are system topics.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -3694,7 +3694,7 @@ private Topic getTopicReference(TopicName topicName) {
             throw e;
         } catch (Exception e) {
             if (e.getCause() instanceof NotAllowedException) {
-                throw new RestException(Status.CONFLICT, e.getCause());
+                throw new RestException(Status.BAD_REQUEST, e.getCause());
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3697]: Interesting, do we through `NotAllowedException` for `pulsar().getBrokerService().getTopicIfExists()`? @congbobo184 Could you please help check how to improve this part? From the method name, we only get the topic ref if the topic exists, but looks like it also create topics.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 1 above)
Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:251]: yes

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 3 above)
Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3697]: ok

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 4]
```diff
@@ -250,7 +250,7 @@ protected void validateAdminAndClientPermission() {
     protected void validateCreateTopic(TopicName topicName) {
         if (isTransactionInternalName(topicName)) {
             log.warn("Try to create a topic in the system topic format! {}", topicName);
```
Comment (hangc0276) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:252]: change the log.warn information to `log.warn("Forbidden to create transaction internal topic: {}" topicName)`  maybe more clear.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 3 above)
Comment (hangc0276) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3697]: +1

**Comment by codelipenghui** [comment]
@congbobo184 Could you please check the comments?

**Review by gaoran10 — APPROVED** [review]
Lgtm

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**